warehouse/src/main/java/org/dromara/hertzbeat/warehouse/store/HistoryTdEngineDataStorage.java

```diff
@@ -231,25 +231,23 @@ public void destroy() {
             hikariDataSource.close();
         }
     }
-    
+
     @Override
     public Map<String, List<Value>> getHistoryMetricData(Long monitorId, String app, String metrics, String metric, String label, String history) {
         String table = app + "_" + metrics + "_" + monitorId;
         String selectSql =  label == null ? String.format(QUERY_HISTORY_SQL, metric, table, history) :
                 String.format(QUERY_HISTORY_WITH_INSTANCE_SQL, metric, table, label, history);
         log.debug(selectSql);
-        Connection connection = null;
         Map<String, List<Value>> instanceValuesMap = new HashMap<>(8);
-        try {
+        try (Connection connection = hikariDataSource.getConnection();
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(selectSql)) {
             if (!serverAvailable) {
                 log.error("\n\t---------------TdEngine Init Failed---------------\n" +
                         "\t--------------Please Config Tdengine--------------\n" +
                         "\t----------Can Not Use Metric History Now----------\n");
                 return instanceValuesMap;
             }
```

Comment:
👍👍 LGTM! hi，suggest put this before the connection.

Reply:
okay, I done the suggestion.